sktime/classification/early_classification/tests/_test_teaser.py

```diff
@@ -35,15 +35,22 @@ def test_teaser_on_unit_test_data():
     final_decisions = np.zeros(10)
 
     X_test = from_nested_to_3d_numpy(X_test)
-    states = None
+    X_test = X_test[indices, :, :]
+
+    ec_idx = np.arange(0, 10)
+    decisions = np.zeros(len(X_test), dtype=bool)
     for i in teaser.classification_points:
-        X = X_test[indices, :, :i]
-        probas, decisions, states = teaser.predict_proba(X, state_info=states)
+        inv_dec = np.invert(decisions)
+        X_test = X_test[inv_dec, :, :]
```

Comment:
It might be cool to have a helper function for this in the base class? 

It should filter X based on the decisions made.

Reply:
Put in a few helper methods for filtering the data in the next commit.